commons/src/main/java/org/infinispan/commons/util/InfinispanCollections.java

```diff
@@ -312,4 +324,20 @@ public int size() {
       }
       return false;
    }
+
+   public static void assertNotNullEntries(Map<?, ?> map, String name) {
+      Objects.requireNonNull(map, () -> "Map '" + name + "' must be non null.");
+      Supplier<String> keySupplier = () -> "Map '" + name + "' contains null key.";
+      Supplier<String> valueSupplier = () -> "Map '" + name + "' contains null value.";
+      map.forEach((k, v) -> {
+         Objects.requireNonNull(k, keySupplier);
+         Objects.requireNonNull(v, valueSupplier);
+      });
+   }
+
+   public static void assertNotNullEntries(Set<?> set, String name) {
```

Comment:
don't wanna be nitpick, but you could replace the ```Set``` by ```Collection``` and make the method generic for everything :)